### PR TITLE
store/tikv: remove useless codes and tests of region_request

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve?

Problem Summary:

Some codes and tests are legacies and it's not suitable for now, so I remove them. 

For example, the canceled store in the `mocktikv` doesn't return the same error as gRPC and the active canceled error is handled out of the `onSendFail` function.

### What is changed and how it works?
What's Changed:

Remove useless codes and tests of region_request

### Check List 

Tests 

- Unit test

### Release note 

- No release note.

### CODE REVIEW

File: store/tikv/region_request.go [hunk 1]
```diff
@@ -542,36 +538,20 @@ func (s *RegionRequestSender) onSendFail(bo *Backoffer, ctx *RPCContext, err err
 	if span := opentracing.SpanFromContext(bo.GetCtx()); span != nil && span.Tracer() != nil {
 		span1 := span.Tracer().StartSpan("regionRequest.onSendFail", opentracing.ChildOf(span.Context()))
 		defer span1.Finish()
-		// TODO(MyonKeminta): Make sure trace works without cloning the backoffer.
-		// bo = bo.Clone()
 		bo.SetCtx(opentracing.ContextWithSpan(bo.GetCtx(), span1))
 	}
-	// If it failed because the context is cancelled by ourself, don't retry.
-	if errors.Cause(err) == context.Canceled {
-		return errors.Trace(err)
```
Comment (lysu) [inline store/tikv/region_request.go:551]: maybe this code is useful.

for example `select a from t use index(idx1) where id = 1`, tidb will fork N goroutine to read region, first goroutine got 1 record that meets "limit 1" will cancel remain N-1 goroutine's context. Context.

for those N-1 goroutine have chance be cancelled in batchclient https://github.com/pingcap/tidb/blob/7c8ddd808aad077e8da436d737860626ea7ffc4b/store/tikv/client_batch.go#L766 and step into this if, it's better to return immediately for those request instead of retry

Comment (youjiali1995) [reply store/tikv/region_request.go:551]: Cancellation is handled at https://github.com/pingcap/tidb/blob/e167b26e11bde210792c53ccddf99ca01293a775/store/tikv/region_request.go#L499-L504

File: store/tikv/region_request.go [hunk 2]
```diff
@@ -227,8 +225,6 @@ func (s *RegionRequestSender) SendReqCtx(
 	if span := opentracing.SpanFromContext(bo.GetCtx()); span != nil && span.Tracer() != nil {
 		span1 := span.Tracer().StartSpan("regionRequest.SendReqCtx", opentracing.ChildOf(span.Context()))
 		defer span1.Finish()
-		// TODO(MyonKeminta): Make sure trace works without cloning the backoffer.
-		// bo = bo.Clone()
 		bo.SetCtx(opentracing.ContextWithSpan(bo.GetCtx(), span1))
```
Comment (MyonKeminta) [inline store/tikv/region_request.go:228]: Did you make sure the trace works well? I'm also not sure if it won't cause any problem that we  wraps too much layers on the bo.ctx. 🤔

Comment (youjiali1995) [reply store/tikv/region_request.go:228]: Yes.
```
MySQL [sbtest]> trace select * from sbtest1 where id = 1;
+-----------------------------------------------------------------------+-----------------+------------+
| operation                                                             | startTS         | duration   |
+-----------------------------------------------------------------------+-----------------+------------+
| trace                                                                 | 14:00:46.549722 | 7.401397ms |
|   ├─session.ExecuteStmt                                               | 14:00:46.549739 | 305.158µs  |
|   │ ├─executor.Compile                                                | 14:00:46.549754 | 34.52µs    |
|   │ └─session.runStmt                                                 | 14:00:46.549872 | 71.825µs   |
|   ├─*executor.PointGetExecutor.Next                                   | 14:00:46.550061 | 6.907209ms |
|   │ ├─tikvSnapshot.get                                                | 14:00:46.550068 | 6.884528ms |
|   │ └─regionRequest.SendReqCtx                                        | 14:00:46.550873 | 758.632µs  |
|   │   ├─rpcClient.SendRequest, region ID: 664100, type: Get           | 14:00:46.550905 | 662.145µs  |
|   │   └─tikv.onRegionError                                            | 14:00:46.551617 | 6.95µs     |
|   │     ├─tikv.backoff.regionMiss                                     | 14:00:46.551694 | 2.228813ms |
|   │     ├─loadRegion                                                  | 14:00:46.554161 | 542.379µs  |
|   │     └─regionRequest.SendReqCtx                                    | 14:00:46.554732 | 2.206014ms |
|   │       └─rpcClient.SendRequest, region ID: 671408, type: Get       | 14:00:46.554742 | 2.170395ms |
|   └─*executor.PointGetExecutor.Next                                   | 14:00:46.556995 | 2.212µs    |
+-----------------------------------------------------------------------+-----------------+------------+
14 rows in set (0.01 sec)
```

**Review by sticnarf — APPROVED** [review]
LGTM

File: store/tikv/region_request.go [hunk 3]
```diff
@@ -542,36 +538,20 @@ func (s *RegionRequestSender) onSendFail(bo *Backoffer, ctx *RPCContext, err err
...
 	}
-	// If it failed because the context is cancelled by ourself, don't retry.
-	if errors.Cause(err) == context.Canceled {
-		return errors.Trace(err)
-	} else if atomic.LoadUint32(&ShuttingDown) > 0 {
+	if atomic.LoadUint32(&ShuttingDown) > 0 {
 		return tikverr.ErrTiDBShuttingDown
 	}
-	if status.Code(errors.Cause(err)) == codes.Canceled {
-		select {
```
Comment (MyonKeminta) [inline store/tikv/region_request.go:556]: Why is this part removed? I didn't understand 😕

Comment (youjiali1995) [reply store/tikv/region_request.go:556]: Do you mean to keep the warning log?

Comment (MyonKeminta) [reply store/tikv/region_request.go:556]: Ah, sorry, I think I got it. Nevermind.

**Review by MyonKeminta — APPROVED** [review]

File: store/tikv/region_request.go (hunk 3 above)
Comment (youjiali1995) [reply store/tikv/region_request.go:556]: About the network error, I added a `TODO` and I will handle it carefully later.

**Comment by youjiali1995** [comment]
@AndreMouche Some tests depend on the mock TiKV returning the `context.Canceled` error, so tests can't pass. I will fix it later.

**Comment by AndreMouche** [comment]
Could we close this PR now? Since store/tikv is moving to github.com/tikv/client-go?
